tap_spreadsheets_anywhere/file_utils.py

```diff
@@ -167,6 +172,53 @@ def get_matching_objects(table_spec, modified_since=None):
         LOGGER.info(f'Processing {len(to_return)} resolved objects that met our criteria. Enable debug verbosity logging for more details.')
     return sorted(to_return, key=lambda item: item['last_modified'])
 
+def evaluate_row_level_date_criteria(target_file_name, table_spec, modified_since=None):
+    protocol, bucket = parse_path(table_spec['path'])
+
+    pattern = table_spec['pattern']
+    matcher = re.compile(pattern)
+
+    to_return = []
+            
+    s3 = boto3.client('s3')
+    
+    resp = s3.select_object_content(
+        Bucket=table_spec['path'].replace("s3://", ""),
+        Key=target_file_name,
+        ExpressionType='SQL',
+        Expression="SELECT CAST(_7 AS TIMESTAMP) as modified_date from s3object",
+        InputSerialization = {'CSV': {"FileHeaderInfo": "NONE"}, 'CompressionType': 'GZIP'},
+        OutputSerialization = {'CSV': {}},
+    )
+        
+    max_date_list = []
+    
+    for event in resp['Payload']:
+
+        if 'Records' in event:    
+            records = event['Records']['Payload'].decode('utf-8')
+            records_list = records.splitlines()
+            dates_list = [parse(date) for date in records_list if not date is None]             
+            max_date_list.append(max(dt for dt in dates_list))
+    latest_modified_record = max(dt for dt in max_date_list)
+    #latest_modified_record = max(dates_list)
```

Comment:
Just small nit - to remove the commented out lines, this one and the below, or if some reason they're in here for a reason, comments about why...

Reply:
Thanks. I will push a change for this line, but the others are from the source repo.